pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -443,6 +472,18 @@ func reportMetrics(pa *pav1alpha1.PodAutoscaler, want int32, got int) error {
 	return nil
 }
 
+func (c *Reconciler) verifyContainer(pa *pav1alpha1.PodAutoscaler) (*apis.Condition, error) {
+	rev, err := c.revLister.Revisions(pa.Namespace).Get(pa.Name)
```

Comment:
I don't like the layering violation that's here.  Revision creates the PodAutoscaler abstraction, so this is a back-edge in the layering graph, which I've been trying to eliminate. :(